File: app/src/main/java/com/checkin/app/checkin/Data/RetrofitCallAsyncTask.java
package com.checkin.app.checkin.Data;

import android.os.AsyncTask;

import com.checkin.app.checkin.Utility.ProgressRequestBody;

import java.io.IOException;

import retrofit2.Call;
import retrofit2.Response;

public class RetrofitCallAsyncTask<T> extends AsyncTask<Call<T>, Void, Void> {
    private ProgressRequestBody.UploadCallbacks mListener;

    public RetrofitCallAsyncTask(ProgressRequestBody.UploadCallbacks listener) {
        mListener = listener;
    }

    @SafeVarargs
    @Override
    protected final Void doInBackground(Call<T>... calls) {
        for (Call call: calls) {
            try {
                Response response = call.execute();
                if (mListener != null) {
                    if (response.isSuccessful())
                        mListener.onSuccess();
                    else
                        mListener.onFailure();
                }
            } catch (IOException e) {
                e.printStackTrace();
                mListener.onFailure();
            }
        }
        return null;
    }
}

File: app/src/main/java/com/checkin/app/checkin/data/network/RetrofitCallAsyncTask.kt
package com.checkin.app.checkin.data.network

import android.os.AsyncTask
import com.checkin.app.checkin.utility.ProgressRequestBody.UploadCallbacks
import retrofit2.Call
import java.io.IOException

class RetrofitCallAsyncTask<T>(private val mListener: UploadCallbacks<T>?) : AsyncTask<Call<T>, Void?, Void?>() {
    @SafeVarargs
    override fun doInBackground(vararg calls: Call<T>): Void? {
        for (call in calls) {
            try {
                val response = call.execute()
                if (response.isSuccessful) mListener?.onSuccess(ApiResponse(response))
                else mListener?.onFailure(ApiResponse(response))
            } catch (e: IOException) {
                e.printStackTrace()
                mListener?.onFailure(ApiResponse(e))
            }
        }
        return null
    }
}